localstack-core/localstack/utils/container_utils/docker_cmd_client.py

```diff
@@ -266,17 +266,21 @@ def commit(
             ) from e
 
     def remove_container(self, container_name: str, force=True, check_existence=False) -> None:
-        if check_existence and container_name not in self.get_running_container_names():
+        if check_existence and container_name not in self.get_all_container_names():
             return
         cmd = self._docker_cmd() + ["rm"]
         if force:
             cmd.append("-f")
         cmd.append(container_name)
         LOG.debug("Removing container with cmd %s", cmd)
         try:
-            run(cmd)
+            output = run(cmd)
+            # When the container does not exist, the output could have the error message without any exception
+            if isinstance(output, str) and not force:
+                self._check_output_and_raise_no_such_container_error(container_name, output=output)
```

Comment:
Very nice catch! It seems the docker CLI does not exit with a non-zero exit code in this case, weird!